src/libraries/Hooks.sol

```diff
@@ -261,16 +261,16 @@ library Hooks {
             if (key.fee.isDynamicFee()) lpFeeOverride = result.parseFee();
 
             // skip this logic for the case where the hook return is 0
-            if (self.hasPermission(BEFORE_SWAP_RETURNS_DELTA_FLAG)) {
-                hookReturn = BeforeSwapDelta.wrap(result.parseReturnDelta());
+            if (self.hasPermission(BEFORE_SWAP_RETURNS_DELTAS_FLAG)) {
+                hookReturn = BeforeSwapDeltas.wrap(result.parseReturnDeltas());
 
                 // any return in unspecified is passed to the afterSwap hook for handling
-                int128 hookDeltaSpecified = hookReturn.getSpecifiedDelta();
+                int128 hookDeltaspecified = hookReturn.getSpecifiedDelta();
 
                 // Update the swap amount according to the hook's return, and check that the swap type doesnt change (exact input/output)
-                if (hookDeltaSpecified != 0) {
+                if (hookDeltaspecified != 0) {
                     bool exactInput = amountToSwap < 0;
-                    amountToSwap += hookDeltaSpecified;
+                    amountToSwap += hookDeltaspecified;
```

Comment:
same with L271, L273, and L268 - they should all still be `hookDeltaSpecified` because it is only pertaining to one delta in the token specified